projects/packages/connection-ui/src/class-initial-state.php

```diff
@@ -35,8 +37,21 @@ public function __construct() {
 	 * @return array
 	 */
 	private function get_data() {
+		global $is_safari;
+
 		return array(
 			'connectionStatus' => REST_Connector::connection_status( false ),
+			'API'              => array(
+				'WP_API_root'       => esc_url_raw( rest_url() ),
+				'WP_API_nonce'      => wp_create_nonce( 'wp_rest' ),
+				'registrationNonce' => wp_create_nonce( 'jetpack-registration-nonce' ),
+			),
+			'connectionData'   => array(
+				'doNotUseConnectionIframe' => $is_safari || User_Agent_Info::is_opera_desktop() || Constants::is_true( 'JETPACK_SHOULD_NOT_USE_CONNECTION_IFRAME' ),
```

Comment:
Should we check the `jetpack_use_iframe_authorization_flow` filter here as well? That filter is described [here](https://github.com/Automattic/jetpack/blob/2cdf4dc27aa4cd98207693d798ff8968fc8876ad/projects/packages/connection/src/class-manager.php#L810).

Reply:
Added, thanks!